tests/cpp/multidevice.cpp

```diff
@@ -82,6 +82,80 @@ at::Tensor MultiDeviceTest::shardTensor(
   return nvfuser::shardTensor(tensor, axis, mesh, device_id);
 }
 
+// testValidate doesn't work out of the box due to #2906, so I had to manually
+// specify the absolute tolerances. The atols passed in are tuned for bfloat,
+// the least precise dtype. They can probably be made stricter for other
+// dtypes.
+void MultiDeviceTest::validate(
```

Comment:
Moving this here to use in the added test.
I made it a class method to restrict usage to multidevice tests.